socket/src/main/java/net/runelite/client/plugins/socket/SocketPlugin.java

```diff
@@ -20,331 +43,164 @@
 import net.runelite.client.plugins.Plugin;
 import net.runelite.client.plugins.PluginDescriptor;
 import net.runelite.client.plugins.PluginType;
-import net.runelite.client.plugins.socket.org.json.JSONException;
+import net.runelite.client.plugins.socket.hash.AES256;
 import net.runelite.client.plugins.socket.org.json.JSONObject;
 import net.runelite.client.plugins.socket.packet.SocketBroadcastPacket;
+import net.runelite.client.plugins.socket.packet.SocketPlayerJoin;
+import net.runelite.client.plugins.socket.packet.SocketPlayerLeave;
 import net.runelite.client.plugins.socket.packet.SocketReceivePacket;
 import org.pf4j.Extension;
+import javax.inject.Inject;
+import java.io.PrintWriter;
 
+@Slf4j
 @Extension
 @PluginDescriptor(
-	name = "Socket",
-	description = "Socket connection for broadcasting messages across clients.",
-	tags = {"socket", "server", "discord", "connection", "broadcast"},
-	enabledByDefault = false,
-	type = PluginType.MISCELLANEOUS
+		name = "Socket",
+		description = "Socket connection for broadcasting messages across clients.",
+		tags = {"socket", "server", "discord", "connection", "broadcast"},
+		enabledByDefault = false,
+		type = PluginType.UTILITY
 )
-
-@Slf4j
-public class SocketPlugin extends Plugin implements Runnable
+public class SocketPlugin extends Plugin
 {
 
+	// Config version changes between updates, hence we use a global variable.
+	public static final String CONFIG_VERSION = "Socket Plugin v2.0.5";
```

Comment:
this should just be SocketPlugin, having a different config version for each revision is awful and will fill up the config alot.